animals/animal.go

```diff
@@ -0,0 +1,50 @@
+package animals
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type Species int
+
+var savedAnimals []Animal
+
+const (
+	Pig Species = iota
+	Horse
+	Cow
+)
+
+var SpeciesNames = [3]string{
+	Pig:   "Pig",
+	Horse: "Horse",
+	Cow:   "Cow",
+}
+
+type Animal struct {
+	Species Species
+	Name    string
+	Age     int
+}
+
+func IndexHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	encoder := json.NewEncoder(w)
+	encoder.Encode(savedAnimals)
+}
+
+func CreateHandler(w http.ResponseWriter, r *http.Request) {
+	animal := new(Animal)
```

Comment:
OK, that works for a specific type of input. But what if we got a request in JSON form instead, like we would do with a JSON API? That would make our handler quite a bit simpler, and it wouldn't probably require responding to the customer at all.